src/Cardano/Wallet/BlockSyncer.hs

```diff
@@ -0,0 +1,158 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Cardano.Wallet.BlockSyncer
+  (
+    blockSyncerBracket
+  , runBlockLayerMockup
+  , BlocksConsumed(..)
+  ) where
+
+import Prelude
+
+
+import Control.Concurrent
+    ( forkIO, killThread, myThreadId, threadDelay )
+import Control.Exception
+    ( SomeException, bracket, catch )
+import Data.IORef
+    ( IORef, newIORef, readIORef, writeIORef )
+import Data.List
+    ( delete, (!!) )
+import Data.Text.Internal
+    ( showText )
+import Data.Word
+    ( Word64 )
+import Formatting
+    ( sformat, (%) )
+import qualified Formatting as F
+
+
+import Cardano.Wallet.Primitive
+    ( Block (..), BlockHeader (..), Hash (..) )
+
+
+
+newtype BlocksConsumed = BlocksConsumed [(Hash "BlockHeader")] deriving (Show, Eq)
+
+data BlockLayerMockup = BlockLayerMockup {
+      getBlock      :: (Hash "BlockHeader") -> IO Block
+    , getEpoch      :: Word64 -> IO [Block]
+    , getNetworkTip :: IO ((Hash "BlockHeader"),BlockHeader)
+    }
+
+runBlockLayerMockup
+    :: Int
+    -> [((Hash "BlockHeader"),Block)]
+    -> Int
+    -> IO BlockLayerMockup
```

Comment:
General remark about function API, but primitive types like `Int` or `String` communicate badly about their usage. It's always a good idea to add a comment explaining what the parameter is for. It's usually an even better idea to create a proper type to encapsulate that semantic.

Reply:
:+1: